### PR TITLE
fix: checking password section in auth block

### PR DESCRIPTION
**Pull request by lukzeg**
fix: checking block for input data in auth to return correct state if all args are properly filled.

add: formatter for project.json configuration file, to do not write sample in one line.

### CODE REVIEW

**Comment by lukzeg** [comment]
Hi @pedroresende,
May I propose to add this sample fix for login and password.
It solve the problem mentioned at: https://github.com/adisreyaj/vscode-sonarqube-status/issues/26

File: src/data/constants.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
   sonarURL: '<your-sonar-url>',
+  auth:
+  {
+    username: '<sonar-username>',
+    password: '<sonar-password>'
+  }
+};
+
+export const VSCODE_PROJECT_JSON_FORMAT_OPTIONS = {
+  spaces: 4,
```
Comment (pedroresende) [inline src/data/constants.ts:12]: Why is this needed ?

Comment (lukzeg) [reply src/data/constants.ts:12]: Hi @pedroresende ,
I added this to format project.json file.
As previosly newly created project.json file have all keys and values in one line. After adding it, newly created project.json file will look:
{
    project: '<your-key-here>',
    sonarURL: '<your-sonar-url>',
    auth:
    {
      username: '<sonar-username>',
      password: '<sonar-password>',
    }
}

**Comment by lukzeg** [comment]
Hi @adisreyaj ,
My apologies for forgetting to add you in previously  comments.

Please let me know if the fix fit to your standards.

Lucas

File: src/data/constants.ts [hunk 2]
```diff
@@ -1,6 +1,16 @@
 export const VSCODE_PROJECT_CONFIG = {
   project: '<your-key-here>',
   sonarURL: '<your-sonar-url>',
+  auth:
```
Comment (adisreyaj) [inline src/data/constants.ts:4]: Need to add support to provide `token` instead of `username` and `password` too. So maybe will add some documentation to let users know about it.

**Review by adisreyaj — APPROVED** [review]

File: src/data/constants.ts (hunk 2 above)
Comment (lukzeg) [reply src/data/constants.ts:4]: Hi

 I done two way of authentication. As on my site part of persons use token or username and password. I will clean up the code and create PR today or tomorrow

Comment (adisreyaj) [reply src/data/constants.ts:4]: https://github.com/adisreyaj/vscode-sonarqube-status/pull/30

I've added token based auth too. Would appreciate if you can try it out locally once.
